### PR TITLE
Light refactor, tidy up, typing

### PR DESCRIPTION
**Pull request by davidbenicek**
check my comments bellow :)

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -39,6 +39,7 @@ module.exports = {
 
         // Why would you want unused vars?
         '@typescript-eslint/no-unused-vars': ['error'],
+        'semi': ["error", "always", { "omitLastInOneLineBlock": true}],
```
Comment (davidbenicek) [inline .eslintrc.js:42]: this and the change bellow - it was deleting my semicolons and driving me mad 😂

File: client/components/SelectableButton.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import styled from 'styled-components';
+
+interface SelectableButtonProps {
+  readonly isSelected: boolean;
+}
+
+const SelectableButton = styled.button<SelectableButtonProps>`
```
Comment (davidbenicek) [inline client/components/SelectableButton.tsx:7]: I made this more generic. Let me know if that's okay

File: client/constants.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+export const LOCAL_STORAGE_KEY = 'weeggets';
+
+export const CATEGORY_RESTAURANT: 'RESTAURANT' = 'RESTAURANT' as const;
+export const CATEGORY_GASTROPUB = 'GASTROPUB' as const;
```
Comment (davidbenicek) [inline client/constants.ts:4]: added some extra ones but we can delete them to simplify if we want

File: components/BusinessTypeButton.tsx [hunk 4]
```diff
@@ -1,17 +0,0 @@
-import styled from 'styled-components'
-
-interface BusinessTypeProps {
-  readonly selected: boolean;
-  readonly value: string;
-};
-
-const StyledButton = styled.button<BusinessTypeProps>`
-  background-color: ${props => props.selected === props.value ? 'green' : 'grey'};
```
Comment (davidbenicek) [inline components/BusinessTypeButton.tsx:9]: Note: this didn't really work because it was comparing a boolean to a string

File: pages/index.tsx [hunk 5]
```diff
@@ -1,85 +1,115 @@
...
 
   useEffect(() => {
-    window.localStorage.setItem(key, JSON.stringify(value))
-  }, [key, value])
+    window.localStorage.setItem(LOCAL_STORAGE_KEY, JSON.stringify(state));
+  }, [state]);
 
-  return [value, setValue]
-}
+  return { state, setValue };
```
Comment (davidbenicek) [inline pages/index.tsx:40]: it was giving me weird typing errors when returning as a tuple in an array. I think this is kind of neater anyway but don't have my mind set on it

File: pages/index.tsx [hunk 6]
```diff
@@ -1,85 +1,115 @@
...
 
   useEffect(() => {
-    const stickyValue = window.localStorage.getItem(key)
+    const stickyValue = window.localStorage.getItem(LOCAL_STORAGE_KEY);
 
     if (stickyValue) {
-      setValue(JSON.parse(stickyValue))
+      setValue(JSON.parse(stickyValue));
     }
-  }, [key])
```
Comment (davidbenicek) [inline pages/index.tsx:26]: key will never change, so no point adding it as a dependency

**Review by davidbenicek — COMMENTED** [review]
Leaving some comments to aid with review :)

**Review by jack994 — APPROVED** [review]
LGTM, 
Very nice job here tidying up and making everything up to standards 👌 
We can merge this into my pr and my pr into master for the time being and then open other PRs to master, otherwise if we keep working here it becomes a mess.

Gonna type a couple of things here we need to remember to do in the next steps for sure:
- Add translations for italian, we should try to always keep them even (eng and ita)
- Add flag to switch language?
- start doing CSS for how we want our product to finally look (first discuss with lorenzo and maybe his friend designer)
- MOBILE! we want this to be Mobile first.
- maybe split the first two pages
- I was thinking we can have a theme selector or something to let the user decide on a couple of things, but that I think can go as a final page (just typing it otherwise I forget, not too important right now)
- Let's have a chat all together when we can

**Pull request merged**